rainier-trace/src/main/scala/com/stripe/rainier/trace/Tracer.scala

```diff
@@ -21,6 +21,13 @@ case class Tracer(compiler: Compiler, gradient: Boolean) {
       compiler.compileTargets(rv.targets, gradient, batchBits)
     Tracer.dump(df.cf)
   }
+
+  def output(rv: RandomVariable[_], batchBits: Int = 1): String = {
```

Comment:
so this returns a string rather than printing it - out of curiosity, where did you find that useful? Maybe we can find a better abstraction that the two-level code duplication here.